artifactory/services/repositories.go

```diff
@@ -0,0 +1,128 @@
+package services
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+const apiRepositories = "api/repositories"
+
+type RepositoriesService struct {
+	client     *jfroghttpclient.JfrogHttpClient
+	ArtDetails auth.ServiceDetails
+}
+
+func NewRepositoriesService(client *jfroghttpclient.JfrogHttpClient) *RepositoriesService {
+	return &RepositoriesService{client: client}
+}
+
+func (rs *RepositoriesService) Get(repoKey string, repoDetails interface{}) error {
```

Comment:
It was defined as generic to allow passing the interface corresponding to the repository the user is fetching (RepositoryDetails, VirtualRepositoryBaseParams, MavenVirtualRepositoryParams, etc).

Reply:
I understand that @RobiNino , but we may be able use a more specific interface, rather than ```interface{}```, to make it easier for developer to identify all the supported implementations of this inteface.

Reply:
@eyalbe4 We can, but I'm not sure it will make it easier to identify all the supported implementations... 
It requires adding a new interface with some getters/setters.
Adding those new getters and setters to all supported types, that will not be used anywhere...
The Get Repo function itself will not use the required interface capabilities, so it might be even more confused to the developer that wants to use the new fixability we are offering here (it might look like that the func will fill up only the fields that appear in the new marker interface).